api/src/flows.ts

```diff
@@ -421,8 +421,12 @@ class FlowManager {
 			let data;
 
 			if (error instanceof Error) {
-				// If the error is instance of Error, use the message of it as the error data
-				data = { message: error.message };
+				// If the error is from a request or exec operation, use the message of it as the error data
+				if (operation.type == 'request' || operation.type == 'exec') {
+					data = { message: error.message };
+				} else {
+					data = error;
+				}
```

Comment:
As the original issue is that the code is not passed on, should we not add that instead. The point of the mentioned PR was to not expose the stack trace but we can add the code where applicable 😄 
```suggestion
if (error instanceof BaseException) {
	// "code" is part of our BaseException
	data = { message: error.message, code: error.code, extensions: error.extensions };
} else if (error instanceof Error) {
	// and does not exist in the native Error
	data = { message: error.message };
```

Reply:
@br41nslug That looks great to me @br41nslug!